http/src/v0/id.rs

```diff
@@ -0,0 +1,91 @@
+use super::{with_ipfs, InvalidPeerId, NotImplemented, StringError};
+use ipfs::{Ipfs, IpfsTypes, PeerId};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use warp::{query, Filter};
+
+pub fn identity<T: IpfsTypes>(
+    ipfs: &Ipfs<T>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    warp::path!("id")
+        .and(with_ipfs(ipfs))
+        .and(optional_peer_id())
+        .and_then(identity_query)
+}
+
+fn optional_peer_id(
+) -> impl Filter<Extract = (Option<PeerId>,), Error = warp::Rejection> + Clone + Copy {
```

Comment:
Not sure why doesn't clippy warn on that `Clone + Copy`. I'll try to make this as a non-async `map` as well. Had some troubles around here because I had forgotten `async` modifier on the `fn identity<T>` which gave an hard to debug error message.

Reply:
`Filter::map` is not able to return Results which makes `and_then` only one usable right here.